### PR TITLE
Bump kube-openapi to latest

### PR DESCRIPTION
**Pull request by jpbetz**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Bump k8s.io/kube-openapi to latest.

Drops dependency on gopkg.in/yaml.v2!

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

/triage accepted
/sig api-machinery

### CODE REVIEW

**Comment by jpbetz** [comment]
/assign @liggitt @dims 

This simplifies our dependencies slightly.

File: vendor/modules.txt [hunk 1]
```diff
@@ -1015,9 +1015,6 @@ gopkg.in/square/go-jose.v2
 gopkg.in/square/go-jose.v2/cipher
 gopkg.in/square/go-jose.v2/json
 gopkg.in/square/go-jose.v2/jwt
-# gopkg.in/yaml.v2 v2.4.0
-## explicit; go 1.15
-gopkg.in/yaml.v2
```
Comment (liggitt) [inline vendor/modules.txt:1020]: 🎉

**Review by liggitt — COMMENTED** [review]
nice! might want to add to hack/unwanted-dependencies.json to keep it from creeping back into vendor (with a comment to prefer `sigs.k8s.io/yaml`)

**Comment by jpbetz** [comment]
> nice! might want to add to hack/unwanted-dependencies.json to keep it from creeping back into vendor (with a comment to prefer `sigs.k8s.io/yaml`)

Handy!  Done.

File: vendor/modules.txt (hunk 1 above)
Comment (aojea) [reply vendor/modules.txt:1020]: yay

Comment (dims) [reply vendor/modules.txt:1020]: Sweet!

**Pull request merged**